apis/marathon.py

```diff
@@ -44,50 +44,33 @@ def create_run():
         return jsonify(id)
 
     return "Error submitting prompt"
-
-"""
-@marathon_api.post('/gen/')
-@check_admin
-def create_marathon_prompt():
-    
-    print("Received marathon prompt req")
-    print(request.json)
     
+@marathon_api.patch('/update_anonymous')
+def update_anonymous_marathon_run():
+    '''
+    Updates the user_id of a given run_id, only if the associated run_id is an anonymous run
+    '''
 
-    nbucket = int(request.json.get("nbucket"))
-    nbatch = int(request.json.get("nbatch"))
-    nperbatch = int(request.json.get("nperbatch"))
-    
-    if nbucket < 1 or nbatch < 1 or nperbatch < 1:
-        raise ValueError("Bad input")
+    query = 'UPDATE `marathonruns` SET `user_id`=%s WHERE `run_id`=%s AND `user_id` IS NULL' 
     
+    if ('user_id' in session):
+        user_id = session['user_id']
+    else:
+        return f'Error updating marathon runs, user not logged in', 500
```

Comment:
`500` isn't the right error code here, we want to reserve this for unexpected errors